### PR TITLE
[onert] Introduce SGD optimizer

### PR DESCRIPTION
**Pull request by ragmani**
This commit introduces SGD optimizer.

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

**Comment by jyoungyun** [comment]
@nnfw-bot test onert-x64-release

File: runtime/onert/core/src/exec/train/optimizer/SGD.cc [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  {
+    throw std::runtime_error("SGD: Invalid gradient tensor");
+  }
+
+  switch (grad_tensor.data_type())
+  {
+    case ir::DataType::FLOAT32:
+      elementwise<float>(shape, grad_tensor, trainable_tensor,
+                         [&](float dst, float src) -> float { return dst + src * lr; });
+      break;
```
Comment (jyoungyun) [inline runtime/onert/core/src/exec/train/optimizer/SGD.cc:57]: (out of curiosity)
Can this lambda function be changed according to data type?

**Review by jyoungyun — APPROVED** [review]
LGTM

**Comment by ragmani** [comment]
@nnfw-bot test onert-x64-release

File: runtime/onert/core/src/exec/train/optimizer/SGD.cc (hunk 1 above)
Comment (ragmani) [reply runtime/onert/core/src/exec/train/optimizer/SGD.cc:57]: Yes, unless the data type is a quantization type.

**Comment by glistening** [comment]
I expected `SGD` will be the simplest optimizer.
But this codes has more like `momentum`, `nesterov`.

After searching more, I realized that it is about `SGD` with momentum (more general version). `SGD` may be set by using your default value:

```cc
double momentum{0.0};
bool nesterov{false};
```

**Review by glistening — APPROVED** [review]
LGTM

**Pull request merged**